tests/sklearn/test_sklearn_training_session.py

```diff
@@ -0,0 +1,50 @@
+import pytest
```

Comment:
Created a file that focuses on `_SklearnTrainingSession` to simplify test function names

Reply:
I'll add more tests

Reply:
Looks great! Thanks so much. On the subject of additional testing, it'd be awesome to include a case verifying that, if we create a session with `allow_children=False` and a nested session with `allow_children=True`, additional nested sessions should still not log (i.e., the top-level session's `allow_children=True` should override the second level's `allow_children=False`).